### PR TITLE
avoid repeated block verification

### PR DESCRIPTION
**Pull request by rlan35**
Verified blocks should always be marked to avoid repeated verification on the same block.

### CODE REVIEW

File: consensus/validator.go [hunk 1]
```diff
@@ -173,11 +173,13 @@ func (consensus *Consensus) onPrepared(recvMsg *FBFTMessage) {
 		consensus.getLogger().Debug().Msg("[onPrepared] consensus received message before init. Ignoring")
 		return
 	}
-	if err := consensus.BlockVerifier(&blockObj); err != nil {
-		consensus.getLogger().Error().Err(err).Msg("[OnPrepared] Block verification failed")
-		return
+	if !consensus.FBFTLog.IsBlockVerified(&blockObj) {
```
Comment (JackyWYX) [inline consensus/validator.go:176]: Since this piece of code is used many times, can we extract it to a function?

**Review by JackyWYX — APPROVED** [review]

**Review by LeoHChen — COMMENTED** [review]
agree with Jacky, it is actually would be better to wrap the checking within blockverifier function to simplify the change.

File: consensus/validator.go (hunk 1 above)
Comment (rlan35) [reply consensus/validator.go]: done.

**Comment by rlan35** [comment]
Travis, test this please.

**Review by LeoHChen — APPROVED** [review]

**Pull request merged**